bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/connector/PowermaxConnector.java

```diff
@@ -106,14 +106,25 @@ public void handleIncomingMessage(byte[] incomingMessage) {
         }
     }
 
+    /**
+     * Handles a communication failure
+     */
+    public void handleCommunicationFailure() {
+        close();
+
+        for (int i = 0; i < listeners.size(); i++) {
+            listeners.get(i).onCommunicationFailure();
+        }
+    }
+
     @Override
     public void sendMessage(byte[] data) {
         try {
             output.write(data);
             output.flush();
         } catch (IOException e) {
-            logger.debug("sendMessage(): Writing error: {}", e.getMessage(), e);
-            setConnected(false);
+            logger.warn("sendMessage(): Writing error: {}", e.getMessage(), e);
```

Comment:
If an error is also set in the status of the thing you should not also log to warn. Maybe pass the message to `handleCommunicationFailure` and set the message in the status. So the user doesn't have to go looking into the log to see the specific message.

Reply:
Ok. I found it odd that all of the binding messages are Debug, which means it's all or nothing... and unless you have the log level set to Debug, you don't even get basic logging to tell you that it's working. I read the guidance in the docs though and I'll make some adjustments.